WordPress/Classes/ViewRelated/Gutenberg/Layout Picker/GutenbergLayoutPickerViewController.swift

```diff
@@ -112,7 +112,24 @@ class GutenbergLayoutPickerViewController: UIViewController, CollapsableHeaderDa
         guard let blog = blog else { return }
         isLoading = resultsController?.isEmpty() ?? true
         let expectedThumbnailSize = LayoutPickerSectionTableViewCell.expectedTumbnailSize
-        PageLayoutService.layouts(forBlog: blog, withThumbnailSize: expectedThumbnailSize)
+        PageLayoutService.layouts(forBlog: blog, withThumbnailSize: expectedThumbnailSize) { [weak self] result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success:
+                    self?.headerContentsDelegate?.dismissNoResultsController()
+                case .failure(let error):
+                    self?.handleErrors(error)
+                }
+            }
+        }
+    }
+
+    private func handleErrors(_ error: Error) {
+        guard resultsController?.isEmpty() ?? true else { return }
```

Comment:
Not suggesting a change here, just curious about the case where there is an error, but `resultsController` is still populated, can such a state be reached (if so, is it unexpected / worth logging)?

Reply:
Good question. The data is cached here so the most common use case would be:
1. A user has a good network connection
1. They create a page 
    - This would fetch and cache the results
1. They then go offline
1. They create a second page
    - This would pull from the cache so `resultsController?.isEmpty()` should return false and the user can select from the stored templates.

Reply:
Ah, that makes sense. Thanks for the explanation :+1: